pandas/tests/indexing/test_coercion.py

```diff
@@ -258,6 +258,13 @@ def test_setitem_series_timedelta64(self, val, exp_dtype):
         )
         self._assert_setitem_series_conversion(obj, val, exp, exp_dtype)
 
+    def test_setitem_series_colon(self):
+        # GH35865 - int casted to str when internally calling np.array(ser.values)
+        ser = pd.Series(["a", 1])
+        ser[:] = list(ser.values)
+
+        assert ser[1] == 1
```

Comment:
please use tm.assert_series_equal and check the whole object.

I think it is enough to link the issue and give the test a bit more of a specific name

Reply:
Do you mean something like this?

```python
def test_setitem_series_colon(self):
    # GH35865 - int casted to str when internally calling np.array(ser.values)
    ser1 = pd.Series(["a", 1])
    ser2 = pd.Series(["a", 1)]
    ser2[:] = list(ser2.values)

    tm.assert_series_equal(ser1, ser2)
```

Also, I'm unsure what an appropriate name would be. I was following the `test_setitem_series` structure from the file. Maybe `test_setitem_series_no_coercion_from_values_list`? Might be too long

Reply:
Also, what do you mean by linking the issue? The number is referenced in the first comment of the test, should I add the full url?

Reply:
Length is ok, name sounds good.

sorry no I meant only adding the gh reference, no additional comment

Reply:
Done!